김은지/오픈채팅방.js

```diff
@@ -0,0 +1,19 @@
+const logMsg = {
+  Enter: (name) => `${name}님이 들어왔습니다.`,
+  Leave: (name) => `${name}님이 나갔습니다.`,
+};
+
+function solution(record) {
+  const members = record.reduce((acc, log) => {
+    const [, id, name] = log.split(" ");
+    if (name) acc[id] = name;
+    return acc;
+  }, {});
+
+  return record
+    .map((log) => {
+      const [command, id] = log.split(" ");
+      return logMsg[command]?.(members[id]);
+    })
+    .filter(Boolean);
+}
```

Comment:
```suggestion
const logMsg = {
  Enter: (name) => `${name}님이 들어왔습니다.`,
  Leave: (name) => `${name}님이 나갔습니다.`,
};

function solution(record) {
  const members = record.reduce((acc, log) => {
    const [, id, name] = log.split(" ");
    if (name) acc.set(id, name);
    return acc;
  }, new Map());

  return record
    .map((log) => {
      const [command, id] = log.split(" ");
      return logMsg[command]?.(members.get(id));
    })
    .filter(Boolean);
}
```
<img width="200" alt="image" src="https://github.com/user-attachments/assets/b9672b57-667c-4102-a4dd-c6a3cd290269">

Map이 살짝 더 빠른 것 같네용!